packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -224,6 +227,25 @@ export function evaluateTemplate(vm: VM, html: Template): VNodes {
     return vnodes;
 }
 
+function validateTemplateVersion(template: Template, vm: VM) {
+    // Validate that the template was compiled with the same version of the LWC compiler used for the runtime engine.
+    // Note this only works in unminified dev mode because it relies on code comments.
+    if (template === defaultEmptyTemplate) {
+        // skip the default template
+        return;
+    }
+    const versionMatcher = template.toString().match(/\/\*LWC compiler v(\d+)\*\/\s*}$/);
+    if (!isNull(versionMatcher)) {
+        const version = parseInt(versionMatcher[1], 10);
+        assert.isTrue(
+            version === COMPILER_VERSION_NUMBER,
+            `Mismatch between compiled template version and runtime engine version. Current engine is v${COMPILER_VERSION_NUMBER}, but ${getComponentTag(
```

Comment:
It feels a little confusing to frame the "compiler version" as "engine version" in the error message. Would it be more accurate to say `currently used compiler version` or `Current engine uses v2`?

Reply:
You're right, this wording can be improved. Let me know what you think of the new version.